recipes/liburing/all/conanfile.py

```diff
@@ -60,49 +56,57 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "liburing is supported only on linux")
 
-    def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-              destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
+    def layout(self):
+        basic_layout(self, src_folder='src')
 
-        self._autotools = AutoToolsBuildEnvironment(self)
-        args = []
-        if self.options.get_safe("with_libc") == False:
-            args.append("--nolibc")
-        self._autotools.configure(args=args)
-        self._autotools.flags.append("-std=gnu99")
-        return self._autotools
-
-    def _patch_sources(self):
-        for data in self.conan_data.get("patches", {}).get(self.version, []):
-            patch(self, **data)
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        def whether(condition):
+            return "" if condition else None
+        tc.update_configure_args({
+            "--nolibc": whether(not self.options.get_safe("with_libc", default=True)),
```

Comment:
IMO, I believe that it'd be better to simplify it, remove that function which is only used once, and make it explicit:

```suggestion
            "--nolibc": None if self.options.get_safe("with_libc") else "",
```

Reply:
Had to change this slightly, the `default=True` is important, since when the option doesn't exist we need to make sure to not pass the flag at all.